src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/ByValueContentsMarshalKindValidator.cs

```diff
@@ -35,34 +37,25 @@ private static ResolvedGenerator ValidateByValueMarshalKind(TypePositionInfo inf
 
             if (info.IsByRef && info.ByValueContentsMarshalKind != ByValueContentsMarshalKind.Default)
             {
-                return generator with
+                return ResolvedGenerator.ResolvedWithDiagnostics(s_forwarder, generator.Diagnostics.Add(new GeneratorDiagnostic.NotSupported(info, context)
```

Comment:
This solution is unintuitive. I say that because the error seemed to be about missing diagnostics but the only difference here seems to be using a difference generator rather than the one passed in. We aren't obviously changing the diagnostic count in this PR. How is this working?

Reply:
The failure is about missing attributes, not diagnostics.

The `Forwarder` is the only generator that will forward parameter attributes. Before this fix, we were using the "selected" generator that would emit the correct marshalling for these error scenarios, and as a result the attributes were not being forwarded. This change moves us back to using the Forwarder for these error scenarios to forward the attributes to the inner DllImport.

Reply:
Ah. Misunderstood the error. Thanks for clarifying.